rllib/core/rl_module/tf/tests/test_tf_rl_module.py

```diff
@@ -95,6 +96,44 @@ def test_get_set_state(self):
         state2_after = module2.get_state()
         check(state, state2_after)
 
+    def test_serialize_deserialize(self):
+        env = gym.make("CartPole-v1")
+        module = DiscreteBCTFModule.from_model_config(
+            env.observation_space,
+            env.action_space,
+            model_config={"hidden_dim": 32},
+        )
+
+        # create a new module from the old module
+        new_module = module.deserialize(module.serialize())
+
+        # check that the new module is the same type
+        self.assertIsInstance(new_module, type(module))
+
+        # check that a parameter of their's is the same
+        self.assertEqual(new_module._input_dim, module._input_dim)
+
+        # check that their states are the same
+        check(module.get_state(), new_module.get_state())
+
+        # check that these 2 objects are not the same object
+        self.assertNotEqual(id(module), id(new_module))
+
+        # check that unpickleable parameters are not allowed by the RL Module
+        # constructor
+        unpickleable_param = threading.Thread()
+
+        def bad_constructor():
+            return DiscreteBCTFModule(
+                input_dim=unpickleable_param,
+                hidden_dim=unpickleable_param,
+                output_dim=unpickleable_param,
+            )
+
+        self.assertRaisesRegex(
```

Comment:
Oh you are also checking the error message with a regex? I found it to make the unittests brittle for no reason when we make error messages more descriptive, just use `self.assertRaises(Exception, bad_constructor)` that's good enough.

Reply:
resolved offline